config/application.rb

```diff
@@ -136,6 +137,16 @@ class Application < Rails::Application
     # Add apidoc hash in headers for smarter caching
     config.middleware.use "Apipie::Middleware::ChecksumInHeaders"
 
+    Foreman::Logging.configure(:log_directory => "#{Rails.root}/log", :environment => Rails.env)
+    Foreman::Logging.add_loggers(SETTINGS[:loggers])
+
+    # Check that the loggers setting exist to configure the app and sql loggers
+    # This is required for upgrades
+    if SETTINGS[:loggers]
```

Comment:
To clarify, create the two loggers app and sql if the config is missing?
On May 21, 2015 7:02 AM, "Dominic Cleal" <notifications@github.com> wrote:

> In config/application.rb
> <https://github.com/theforeman/foreman/pull/2376#discussion_r30791852>:
>
> > @@ -136,6 +137,16 @@ class Application < Rails::Application
> >      # Add apidoc hash in headers for smarter caching
> >      config.middleware.use "Apipie::Middleware::ChecksumInHeaders"
> >
> > +    Foreman::Logging.configure(:log_directory => "#{Rails.root}/log", :environment => Rails.env)
> > +    Foreman::Logging.add_loggers(SETTINGS[:loggers])
> > +
> > +    # Check that the loggers setting exist to configure the app and sql loggers
> > +    # This is required for upgrades
> > +    if SETTINGS[:loggers]
>
> Could we do this automatically if the config's missing?
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/theforeman/foreman/pull/2376/files#r30791852>.
>